notify/impl.go

```diff
@@ -254,19 +254,32 @@ func (n *Email) auth(mechs string) (smtp.Auth, error) {
 
 // Notify implements the Notifier interface.
 func (n *Email) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
-	// Connect to the SMTP smarthost.
-	c, err := smtp.Dial(n.conf.Smarthost)
-	if err != nil {
-		return true, err
-	}
-	defer c.Quit()
-
 	// We need to know the hostname for both auth and TLS.
-	host, _, err := net.SplitHostPort(n.conf.Smarthost)
+	var c *smtp.Client
+	host, port, err := net.SplitHostPort(n.conf.Smarthost)
 	if err != nil {
 		return false, fmt.Errorf("invalid address: %s", err)
 	}
 
+	if port == "465" {
+		conn, err := tls.Dial("tcp", n.conf.Smarthost, &tls.Config{ServerName: host})
+		if err != nil {
+			return true, err
```

Comment:
AFAIK, true/false flag is tied to retry logic (whether the error is recoverable or not). For the communications happening over wire, multiple attempts shall be made(i.e. error shall be deemed to be recoverable, in this case we are trying to establish a connection to remote smtp server). Infact, even before my change most of the error blocks returned true.